### PR TITLE
feat(Themes): Core + Platform Themes

### PR DESCRIPTION
**Pull request by codecaaron**
## Platform Theme Separation - Part 2
- [Part 1](https://github.com/Codecademy/client-modules/pull/1754)
- [Part 2](https://github.com/Codecademy/client-modules/pull/1755)
- [Part 3](https://github.com/Codecademy/client-modules/pull/1756)

Monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/24393

This introduces separate themes for core and platform through gamut-styles and fixes a bunch of the existing behavior to allow for themes to be extended.

This also requires us to update the way we declare the global emotion theme, removing the implicit Core declaration.   This ensures consuming apps will be allowed to declare their own, and all system props types will be derived from the explicit declaration.

As these themes are fully self contained now I have added snapshot tests to ensure that we do not accidentally lose tokens that might break behavior silently.

#### Core Theme
- It includes our basic palette and general sets of tokens.
- Gamut components are only aware of this theme and gamut will compile with reference to this themes types alone.
- Gamut components / system props while within gamut are limited to this, to ensure interoperability between extensions of this base theme.

#### Platform Theme
- Adds a set of net new colors for the learning environment.
- Cannot be consumed within Gamut (shared color mode aliases will work however :)

### PR Checklist

- [x] Related to designs: [Figma](https://www.figma.com/file/QplOE1d1rHbQLu0PtvsWLX/Colors-%E2%9C%85?node-id=487%3A2)
- [x] Related to JIRA ticket: https://codecademy.atlassian.net/browse/GM-174 https://codecademy.atlassian.net/browse/GM-228
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.08eadf.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.08eadf.0<br/>@codecademy/gamut-kit@0.3.101-alpha.08eadf.0<br/>@codecademy/gamut-labs@16.0.6-alpha.08eadf.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.08eadf.0<br/>@codecademy/gamut-styles@10.0.1-alpha.08eadf.0<br/>@codecademy/gamut-tests@2.3.31-alpha.08eadf.0<br/>@codecademy/gamut@32.18.8-alpha.08eadf.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.08eadf.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.08eadf.0<br/>@codecademy/styleguide@37.5.9-alpha.08eadf.0<br/>@codecademy/variance@0.11.3-alpha.08eadf.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.dc7eac.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.dc7eac.0<br/>@codecademy/gamut-kit@0.3.101-alpha.dc7eac.0<br/>@codecademy/gamut-labs@16.0.6-alpha.dc7eac.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.dc7eac.0<br/>@codecademy/gamut-styles@10.0.1-alpha.dc7eac.0<br/>@codecademy/gamut-tests@2.3.31-alpha.dc7eac.0<br/>@codecademy/gamut@32.18.8-alpha.dc7eac.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.dc7eac.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.dc7eac.0<br/>@codecademy/styleguide@37.5.9-alpha.dc7eac.0<br/>@codecademy/variance@0.11.3-alpha.dc7eac.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609de4d9b9337b481b666e8e--gamut-preview.netlify.app>https://609de4d9b9337b481b666e8e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609de4d9b9337b481b666e8e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.0e043b.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.0e043b.0<br/>@codecademy/gamut-kit@0.3.101-alpha.0e043b.0<br/>@codecademy/gamut-labs@16.0.6-alpha.0e043b.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.0e043b.0<br/>@codecademy/gamut-styles@10.0.1-alpha.0e043b.0<br/>@codecademy/gamut-tests@2.3.31-alpha.0e043b.0<br/>@codecademy/gamut@32.18.8-alpha.0e043b.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.0e043b.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.0e043b.0<br/>@codecademy/styleguide@37.5.9-alpha.0e043b.0<br/>@codecademy/variance@0.11.3-alpha.0e043b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609de7bccd15b14414ccfab5--gamut-preview.netlify.app>https://609de7bccd15b14414ccfab5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609de7bccd15b14414ccfab5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.20e976.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.20e976.0<br/>@codecademy/gamut-kit@0.3.101-alpha.20e976.0<br/>@codecademy/gamut-labs@16.0.6-alpha.20e976.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.20e976.0<br/>@codecademy/gamut-styles@10.0.1-alpha.20e976.0<br/>@codecademy/gamut-tests@2.3.31-alpha.20e976.0<br/>@codecademy/gamut@32.18.8-alpha.20e976.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.20e976.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.20e976.0<br/>@codecademy/styleguide@37.5.9-alpha.20e976.0<br/>@codecademy/variance@0.11.3-alpha.20e976.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609ded02eafbe451ebc184d6--gamut-preview.netlify.app>https://609ded02eafbe451ebc184d6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609ded02eafbe451ebc184d6>Deploy Logs</a></p>

File: packages/gamut-labs/src/AppHeaderMobile/AppHeaderLinkMobile/index.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ type AppHeaderLinkButtonProps = { topSeparator: boolean };
 
 const SeparatorOuter = styled(Box)<AppHeaderLinkButtonProps>`
   border-top: ${({ theme, topSeparator }) =>
-    topSeparator ? `1px solid ${theme.colors['gray-400']}` : ''};
+    topSeparator ? `1px solid ${theme.colors['gray-600']}` : ''};
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/AppHeaderMobile/AppHeaderLinkMobile/index.tsx:24]: Did these sneak through the splitting out? 😄

File: packages/gamut-patterns/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,11 @@
 {
   "extends": "../../tsconfig.json",
-  "include": ["../../typings/*.d.ts", "./src/**/*.ts", "./src/**/*.tsx"],
+  "include": [
+    "../../typings/*.d.ts",
+    "./src/**/*.ts",
+    "./src/**/*.tsx",
+    "../../packages/gamut-styles/dist/typings/theme.d.ts"
```
Comment (JoshuaKGoldberg) [inline packages/gamut-patterns/tsconfig.json:7]:
```suggestion
    "../gamut-styles/dist/typings/theme.d.ts"
```

File: packages/gamut-styles/package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
   },
   "dependencies": {
     "@codecademy/gamut-system": "^0.7.2",
+    "polished": "^4.1.1",
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/package.json:21]:
```suggestion
    "polished": "^4.1.2",
```

**Review by JoshuaKGoldberg — APPROVED** [review]
LOVE this snapshot test. 👌

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.edd35e.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.edd35e.0<br/>@codecademy/gamut-kit@0.3.101-alpha.edd35e.0<br/>@codecademy/gamut-labs@16.0.6-alpha.edd35e.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.edd35e.0<br/>@codecademy/gamut-styles@10.0.1-alpha.edd35e.0<br/>@codecademy/gamut-tests@2.3.31-alpha.edd35e.0<br/>@codecademy/gamut@32.18.8-alpha.edd35e.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.edd35e.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.edd35e.0<br/>@codecademy/styleguide@37.5.9-alpha.edd35e.0<br/>@codecademy/variance@0.11.3-alpha.edd35e.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.355a86.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.355a86.0<br/>@codecademy/gamut-kit@0.3.104-alpha.355a86.0<br/>@codecademy/gamut-labs@17.1.2-alpha.355a86.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.355a86.0<br/>@codecademy/gamut-styles@10.0.1-alpha.355a86.0<br/>@codecademy/gamut-tests@2.3.31-alpha.355a86.0<br/>@codecademy/gamut@32.18.9-alpha.355a86.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.355a86.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.355a86.0<br/>@codecademy/styleguide@38.1.2-alpha.355a86.0<br/>@codecademy/variance@0.11.3-alpha.355a86.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a2e205c5723126020be2b0--gamut-preview.netlify.app>https://60a2e205c5723126020be2b0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a2e205c5723126020be2b0>Deploy Logs</a></p>

**Pull request merged**